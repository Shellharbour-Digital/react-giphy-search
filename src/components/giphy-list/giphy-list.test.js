import { mount } from 'enzyme';
import React from 'react';
import GiphyList from './giphy-list.component';

describe('GiphyList component', () => {
  const mockPropsSuccess = {
    meta: {
      msg: 'mockMsg',
      status: 200
    },
    gifs: [{
      id: 'mockId',
      title: 'mockTitle',
      url: 'www.mockurl.com',
      images: {
        fixed_width: {
          url: 'www.mockurlfixedwidth.com',
          height: 200
        }
      }
    }]
  };

  const mockPropsFailure = {
    meta: {
      msg: 'mockMsg',
      status: 500
    }
  };

  const mockPropsNoResults = {
    meta: {
      msg: 'mockMsg',
      status: 200
    },
    gifs: []
  };

  let wrapper;

  it('expect to render GiphyList component', ()  => {
    wrapper = mount(<GiphyList {...mockPropsSuccess} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('expect to show error if status !==200', () => {
    wrapper = mount(<GiphyList {...mockPropsFailure} />);
    expect(wrapper.exists('GifMessage')).toBe(true);
  });

  it('expect to show no results message', () => {
    wrapper = mount(<GiphyList {...mockPropsNoResults} />);
    expect(wrapper.exists('GifMessage')).toBe(true);
  })
});
