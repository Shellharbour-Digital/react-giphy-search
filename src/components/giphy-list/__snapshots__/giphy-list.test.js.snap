// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiphyList component expect to render GiphyList component 1`] = `
<WithGifs
  gifs={
    Array [
      Object {
        "id": "mockId",
        "images": Object {
          "fixed_width": Object {
            "height": 200,
            "url": "www.mockurlfixedwidth.com",
          },
        },
        "title": "mockTitle",
        "url": "www.mockurl.com",
      },
    ]
  }
  meta={
    Object {
      "msg": "mockMsg",
      "status": 200,
    }
  }
>
  <GiphyList
    gifs={
      Array [
        Object {
          "id": "mockId",
          "images": Object {
            "fixed_width": Object {
              "height": 200,
              "url": "www.mockurlfixedwidth.com",
            },
          },
          "title": "mockTitle",
          "url": "www.mockurl.com",
        },
      ]
    }
    meta={
      Object {
        "msg": "mockMsg",
        "status": 200,
      }
    }
    pagination={Array []}
  >
    <styled.div>
      <div
        className="sc-gKckTs eOXWWN"
      >
        <styled.div>
          <div
            className="sc-iCfLBT KMWCh"
          >
            <GiphyItem
              alt="mockTitle"
              height={200}
              key="mockId"
              src="www.mockurlfixedwidth.com"
              url="www.mockurl.com"
            >
              <styled.div>
                <div
                  className="sc-dkPtyc hdAHYd"
                >
                  <LoadingContainer
                    height={200}
                  >
                    <div
                      className="sc-jRQAMF jWpOkq"
                      height={200}
                    >
                      <Spinner>
                        <styled.div>
                          <div
                            className="sc-bdvvaa loGtTo"
                          >
                            <styled.div>
                              <div
                                className="sc-gsDJrp cDvfmb"
                              />
                            </styled.div>
                          </div>
                        </styled.div>
                      </Spinner>
                    </div>
                  </LoadingContainer>
                  <styled.a
                    href="www.mockurl.com"
                    target="_blank"
                  >
                    <a
                      className="sc-eCImvq PnkkR"
                      href="www.mockurl.com"
                      target="_blank"
                    >
                      <GifImg
                        alt="mockTitle"
                        onLoad={[Function]}
                        src="www.mockurlfixedwidth.com"
                        style={
                          Object {
                            "display": "none",
                          }
                        }
                      >
                        <img
                          alt="mockTitle"
                          className="sc-hKwCoD fYRrVq"
                          onLoad={[Function]}
                          src="www.mockurlfixedwidth.com"
                          style={
                            Object {
                              "display": "none",
                            }
                          }
                        />
                      </GifImg>
                    </a>
                  </styled.a>
                </div>
              </styled.div>
            </GiphyItem>
          </div>
        </styled.div>
      </div>
    </styled.div>
  </GiphyList>
</WithGifs>
`;
